site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5040,6 +5040,46 @@ private function deleteGradedComponent(GradedComponent $graded_component) {
         $this->course_db->query($query, $params);
     }
 
+    private function updateOverallComments(TaGradedGradeable $ta_graded_gradeable) {
+        foreach ($ta_graded_gradeable->getOverallComment() as $user_id => $comment) {
+            $this->updateOverallComment($ta_graded_gradeable, $user_id, $comment);
+        }
+    }
+
+    private function updateOverallComment(TaGradedGradeable $ta_graded_gradeable, $grader_id, $comment) {
+        $g_id = $ta_graded_gradeable->getGradedGradeable()->getGradeable()->getId();
+        $user_id = null;
+        $team_id = null;
+
+        // TODO: replace this with a single upsert when postgres can do an on conflict on
+        //   multiple constraints (gradeable_overall_comment_user_unique, gradeable_overall_comment_team_unique)
+        if ($ta_graded_gradeable->getGradedGradeable()->getGradeable()->isTeamAssignment()) {
+            $team_id = $ta_graded_gradeable->getGradedGradeable()->getSubmitter()->getId();
+            $query = "
+            INSERT INTO gradeable_overall_comment (g_id, goc_user_id, goc_team_id, goc_grader_id, goc_overall_comment)
+                VALUES (?, ?, ?, ?, ?)
+                ON CONFLICT (g_id, goc_team_id, goc_grader_id)
+                DO
+                    UPDATE SET 
+                        goc_overall_comment=?;
+            ";
+        }
+        else {
+            $user_id = $ta_graded_gradeable->getGradedGradeable()->getSubmitter()->getId();
+            $query = "
+            INSERT INTO gradeable_overall_comment (g_id, goc_user_id, goc_team_id, goc_grader_id, goc_overall_comment)
+                VALUES (?, ?, ?, ?, ?)
+                ON CONFLICT (g_id, goc_user_id, goc_grader_id)
+                DO
+                    UPDATE SET 
+                        goc_overall_comment=?;
+            ";
+        }
```

Comment:
Can we use string interpolation instead of having the upsert string for the two cases? This would require the upsert string to only appear once for the small difference of one column in the constraint.

Reply:
I like that. Updated.